pkg/artifactory/datasource_artifactory_file_test.go

```diff
@@ -1,58 +1,169 @@
+			},
 		},
+	})
+}
+
+/*
+Negative test case on file download skip
+When file is present at output_path, checksum of files at output_path & repository path matches
+artifactory_file datasource will skip the download.
+*/
+func TestFileDownloadSkipCheck(t *testing.T) {
+	downloadPath := fmt.Sprintf("%s/crash.zip", t.TempDir())
+	localFileModTime := time.Time{}
+
+	// every instance of RT has this repo and file out-of-the-box
+	const script = `
+		data "artifactory_file" "example" {
+		  repository      = "example-repo-local"
+		  path            = "crash.zip"
+		  output_path     = "%s"
+		  force_overwrite = false
+		}
+	`
+
+	var skipDownloadCheck = func(state *terraform.State) error {
+		download := state.Modules[0].Resources["data.artifactory_file.example"].Primary.Attributes["output_path"]
+		downloadedFileStat, err := os.Stat(download)
+		if err != nil {
+			return err
+		}
+		downloadedFileModTime := downloadedFileStat.ModTime()
+		if downloadedFileModTime.After(localFileModTime) { //determines fresh download occurred during the test step
+			return fmt.Errorf("fresh download observed. Existing file modification time: %v. Fresh downloaded file modification time: %v", localFileModTime, downloadedFileModTime)
+		}
+		return nil
+	}
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          downloadPreCheck(t, downloadPath, &localFileModTime),
 		ProviderFactories: testAccProviders,
 		Steps: []resource.TestStep{
 			{
-				Config: script,
-				Check: func(state *terraform.State) error {
-					download := state.Modules[0].Resources["data.artifactory_file.example"].Primary.Attributes["output_path"]
-					_, err := os.Stat(download)
-					if err != nil {
-						return err
-					}
-					verified, err := VerifySha256Checksum(download, "7a2489dd209d0acb72f7f11d171b418e65648b9cc96c6c351e00e22551fdd8f1")
-
-					if !verified {
-						return fmt.Errorf("%s checksum does not have expected checksum", download)
-					}
-					return err
-				},
+				Config: fmt.Sprintf(script, downloadPath),
+				Check:  skipDownloadCheck,
 			},
 		},
 	})
 }
+
+//Creates new directory tree if not exist
+func createNewDir(srcPath string) error {
+	_, err := os.Stat(srcPath)
+	if err != nil {
+		return nil // path exists, nothing to do
+	}
+	if !errors.Is(err, os.ErrNotExist) {
+		return err // some other kind of non-recoverable error
+	}
+	return os.MkdirAll(srcPath, os.ModePerm)
+}
+
+//Copies file from source path to destination path
+func copyFile(destPath string, srcPath string) error {
+	destDir := filepath.Dir(destPath)
+	err := createNewDir(destDir)
+	if err != nil {
+		return err
+	}
+	fin, err := os.Open(srcPath)
+	if err != nil {
+		return err
+	}
+	defer fin.Close()
+
+	fout, err := os.Create(destPath)
+	if err != nil {
+		return err
+	}
+	defer fout.Close()
+
+	_, err = io.Copy(fout, fin)
+	if err != nil {
+		return err
+	}
+	/*currenttime := time.Now().Local()
+	err = os.Chtimes(destPath, currenttime, currenttime)
+	if err != nil {
```

Comment:
What's this about?

Reply:
file copy not changing modification time. Therefore I am forcing the change. So that it can be compared at later stage.

Reply:
Ok, well get rid of comments

Reply:
will update after few more testing

Reply:
due to some issue I had to force `os.Chtimes`. Later I found the issue and commented `os.Chtimes`. Now commented section removed.